### PR TITLE
Add setting: Use splashscreen as home background

### PR DESCRIPTION
**Pull request by 1hitsong**
**Changes**
Adds new user setting allowing the user to use their generated splashscreen as their home screen background.

### CODE REVIEW

File: settings/settings.json [hunk 1]
```diff
@@ -16,6 +16,19 @@
...
+                "children": [
+                    {
+                        "title": "Use Splashscreen as Home Background",
+                        "description": "Use generated splashscreen image as Jellyfin home background. Jellyfin will need to be closed and reopened for change to take affect.",
+                        "settingName": "ui.home.splashBackground",
+                        "type": "bool",
+                        "default": "false"
+                    }
+                ]
+            },
```
Comment (jimdogx) [inline settings/settings.json:31]: Don't forget to add these new strings to a locale file (i.e. en-us).

**Review by jimdogx — CHANGES_REQUESTED** [review]

File: settings/settings.json (hunk 1 above)
Comment (1hitsong) [reply settings/settings.json:31]: :+1:

**Comment by crobibero** [comment]
What is displayed if the splashscreen isn't available?

**Comment by 1hitsong** [comment]
@crobibero If the splashscreen doesn't load, then the user sees the default grey background. Same as if the setting were disabled.

**Comment by jimdogx** [comment]
Since the Branding/Splashscreen endpoint doesn't appear to be used in any clients yet, I uploaded an image using Insomnia but when I run the Roku client it doesn't use the newly updated image.

**Comment by crobibero** [comment]
Does the endpoint return the new image?

**Comment by jimdogx** [comment]
> Does the endpoint return the new image?

Yes

**Comment by jimdogx** [comment]
Tested this morning and this is working now??!!??!!  I still get the following server side, but it doesn't appear to affect the client:

`System.NotImplementedException: The method or operation is not implemented.
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.ConvertImage(String inputPath, String outputPath) in /Users/jimi/Source/jellyfin-server/MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs:line 822
   at Emby.Drawing.ImageProcessor.GetSupportedImage(String originalImagePath, DateTime dateModified) in /Users/jimi/Source/jellyfin-server/Emby.Drawing/ImageProcessor.cs:line 466`

File: components/home/Home.brs [hunk 2]
```diff
@@ -1,6 +1,11 @@
 sub init()
     m.top.overhangTitle = "Home"
     m.top.optionsAvailable = true
+
+    if get_user_setting("ui.home.splashBackground") = "true"
+        m.backdrop = m.top.findNode("backdrop")
+        m.backdrop.uri = buildURL("/Branding/Splashscreen?format=jpg&foregroundLayer=0.15&fillWidth=1280&width=1280&fillHeight=720&height=720&tag=splash")
+    end if
```
Comment (jimdogx) [inline components/home/Home.brs:8]: Nothing to change for this PR, but the foregroundLayer might be a good user setting.  The particular background I selected for testing shows up too dark for my taste and I'd want to adjust it to be more visible 😄

**Review by jimdogx — APPROVED** [review]

**Comment by Neoakb** [comment]
Is the splashscreen img generated per user, or per server? Is there a way for me (the server admin) to have a "custom" img to use as the background instead of a generated splashscreen?

Example that I use for the webUI

image

I tried replacing /config/data/splashscreen.png with that same image for the WebUI that I'm using, but it's too big. I tried resizing it in PS but couldn't get the image to update within the app, both main and dev app versions.

If also possible, could this setting be defaulted to "on"?

Also, could this setting be implemented in other clients without too much pain? I understand you who work on this client might not work on others, I'm just looking in terms of general, client approved, background customization. 

I'd love to play around with this myself as I know a tiny bit about coding, but I don't know how helpful I would be. 

Maybe there could be a small logo like img (a smallish transparent png?) that could be generated into the background? I'll look more into how the splashscreen is generated.

**Pull request merged**